src/js/views/scores.js

```diff
@@ -5,69 +5,73 @@ define('views/scores',[
     'angular'
 ],function(log) {
     var moduleName = 'scores';
-    return angular.module(moduleName,[]).controller(moduleName+'Ctrl',[
+    return angular.module(moduleName,['filters']).controller(moduleName+'Ctrl',[
         '$scope', '$scores','$teams','$stages','$window',
         function($scope,$scores,$teams,$stages,$window) {
             log('init scores ctrl');
 
             $scope.sort = 'index';
             $scope.rev = true;
 
-            $scope.scores = $scores.scores;
-            $scope.stages = $stages.stages;
+            function enrich(scores) {
+                return scores.map(score => {
+                    score.team = $teams.get(score.teamNumber);
+                    score.stage = $stages.get(score.stageId);
+                    return score;
+                });
+            }
+
+            $scores.init().then(function() {
+                $scope.scores = enrich($scores.scores);
+                $scope.stages = $stages.stages;
```

Comment:
Same response. $scores.init initialize stages. Do you want me to run $stages.init().then(() => $scores.init())? Seems unnecessary

Reply:
Yes, or you can do them 'in parallel' if you like, e.g.:

```
$scores.init();
$stages.init().then(() => $scope.stages = $stages.stages);
```

(Note that the order of the two lines doesn't matter)

It's very necessary, when people start refactoring code. You don't want to start 'fixing' code everywhere because you just changed some internal implementation detail somewhere.